packages/client/src/comp/states/NorthCarolina.tsx

```diff
@@ -32,17 +24,19 @@ export const NorthCarolina = () => {
     }
   }
 
-  return <SignatureBase<NorthCarolinaInfo>enrichValues={enrichValues}>
+  return <SignatureBase<NorthCarolinaInfo> enrichValues={enrichValues}>
     <p>North Carolina requires voters to confirm their identify using one of the following types of identification</p>
-    <Select ref={idTypeRef} label='Identification Type' defaultValue='Select' {...{required: true}}>
-      <Option key={0} hidden={true}/>
-      {[...northCarolinaIdentityType].sort().map((value, key) => {
-        return <Option value={value} key={key+1} label={value}/>
-      })}
-    </Select>
+    <Select
+      label='Identification Type'
+      options={[...northCarolinaIdentityType]}
+      value={idType}
+      onChange={v => setIdType(v)}
+      {...{required: true}}
+    />
     <BaseInput
       id='identityData'
-      ref={idDataRef}
+      value={idData}
+      onChange={e => setIdData(e.currentTarget.value)}
       label='Identity Information'
       pattern='\d\d\d\d+'
```

Comment:
Can we use `pattern` to verify a controlled component like this?  Also, we should use the regex `\d{4,}` -- if it's valid in HTML

Reply:
Thanks, did the changes.

We can use pattern with controlled components, forms will trigger browsers' default 'Fill with the required format' tool tip. Also `\d{4,}` works on HTML (everything works except `/g` `/m` `/i` flags)